purchase_sale_inter_company/models/purchase_order.py

```diff
@@ -81,16 +88,28 @@ def _inter_company_create_sale_order(self, dest_company):
                     "purchase price list currency."
                 )
             )
+        # if a sale order has already been generated
+        # delete it and force the same number
+        force_number = False
+        if self.intercompany_sale_id and self.intercompany_sale_id.state in [
+            "draft",
+            "cancel",
+        ]:
+            force_number = self.intercompany_sale_id.name
+            self.intercompany_sale_id.with_context(force_delete=True).unlink()
```

Comment:
I am not sure about this.
If a sale order already have a picking and you cancel it. I pretty sure you can not delete the sale.

Maybe we should add a try except and if we can unlink it, we can keep the same number. If not we should use a new number as sale name should be uniq